src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/TagletWriterImpl.java

```diff
@@ -363,6 +375,320 @@ private boolean isLongOrHasComma(Content c) {
+                LinkTree lt = (LinkTree) see;
+                var linkRef = lt.getReference();
+                if (linkRef == null) {
+                    messages.warning(ch.getDocTreePath(see), "doclet.link.no_reference");
+                    return invalidTagOutput(resources.getText("doclet.tag.invalid_input", lt.toString()),
+                            Optional.empty());
+                }
+                refSignature = ch.getReferencedSignature(linkRef);
+                label = lt.getLabel();
+            }
+
+            case SEE -> {
+                List<? extends DocTree> ref = ((SeeTree) see).getReference();
+                assert !ref.isEmpty();
+                DocTree ref0 = ref.get(0);
+                switch (ref0.getKind()) {
+                    case TEXT, START_ELEMENT -> {
+                        // @see "Reference"
+                        // @see <a href="...">...</a>
+                        return htmlWriter.commentTagsToContent(element, ref, false, false);
+                    }
+                    case REFERENCE -> {
+                        // @see reference label...
+                        refSignature = ch.getReferencedSignature(ref0);
+                        label = ref.subList(1, ref.size());
+                    }
+                    case ERRONEOUS -> {
+                        return invalidTagOutput(resources.getText("doclet.tag.invalid_input",
+                                        ref0.toString()),
+                                Optional.empty());
+                    }
+                    default -> throw new IllegalStateException(ref0.getKind().toString());
+                }
+            }
+
+            default -> throw new IllegalStateException(kind.toString());
+        }
+
+        return linkSeeReferenceToContent(element,
+                see,
+                refSignature,
+                ch.getReferencedElement(see),
+                ch, tagName,kind == LINK_PLAIN,
+                label,
+                (key, args) -> messages.warning(ch.getDocTreePath(see), key, args)
+        );
+    }
+
+    Content linkSeeReferenceToContent(Element holder, DocTree refTree, String refSignature, Element ref,
+                                      CommentHelper ch, String tagName, boolean isLinkPlain,
+                                      List<? extends DocTree> label,
+                                      BiConsumer<String, Object[]> reportWarning) {
+        Content labelContent = plainOrCode(isLinkPlain,
+                htmlWriter.commentTagsToContent(holder, label, context));
+
+        // The signature from the @see tag. We will output this text when a label is not specified.
+        Content text = plainOrCode(isLinkPlain,
+                Text.of(Objects.requireNonNullElse(refSignature, "")));
+
+        TypeElement refClass = ch.getReferencedClass(ref);
+        Element refMem =       ch.getReferencedMember(ref);
+        String refMemName =    ch.getReferencedMemberName(refSignature);
```

Comment:
Because there may not be a node when the method is called from the snippet world.

See this line in the doc comment:

    * @param refTree the tree node containing the information, or {@code null} if not avaiable